pygmt/datatypes/header.py

```diff
@@ -0,0 +1,247 @@
+        # Name of generating command
+        ("command", ctp.c_char * GMT_GRID_COMMAND_LEN320),
+        # Comments for this data set
+        ("remark", ctp.c_char * GMT_GRID_REMARK_LEN160),
+        # Below are items used internally by GMT
+        # Number of data points (n_columns * n_rows) [paddings are excluded]
+        ("nm", ctp.c_size_t),
+        # Actual number of items (not bytes) required to hold this grid (mx * my),
+        # per band (for images)
+        ("size", ctp.c_size_t),
+        # Bits per data value (e.g., 32 for ints/floats; 8 for bytes).
+        # Only used for ERSI ArcInfo ASCII Exchange grids.
+        ("bits", ctp.c_uint),
+        # For complex grid.
+        # 0 for normal
+        # GMT_GRID_IS_COMPLEX_REAL = real part of complex grid
+        # GMT_GRID_IS_COMPLEX_IMAG = imag part of complex grid
+        ("complex_mode", ctp.c_uint),
+        # Grid format
+        ("type", ctp.c_uint),
+        # Number of bands [1]. Used with GMT_IMAGE containers
+        ("n_bands", ctp.c_uint),
+        # Actual x-dimension in memory. mx = n_columns + pad[0] + pad[1]
+        ("mx", ctp.c_uint),
+        # Actual y-dimension in memory. my = n_rows + pad[2] + pad[3]
+        ("my", ctp.c_uint),
+        # Paddings on west, east, south, north sides [2,2,2,2]
+        ("pad", ctp.c_uint * 4),
+        # Three or four char codes T|B R|C S|R|S (grd) or B|L|P + A|a (img)
+        # describing array layout in mem and interleaving
+        ("mem_layout", ctp.c_char * 4),
+        # Missing value as stored in grid file
+        ("nan_value", gmt_grdfloat),
+        # 0.0 for gridline grids and 0.5 for pixel grids
+        ("xy_off", ctp.c_double),
+        # Referencing system string in PROJ.4 format
+        ("ProjRefPROJ4", ctp.c_char_p),
+        # Referencing system string in WKT format
+        ("ProjRefWKT", ctp.c_char_p),
+        # Referencing system EPSG code
+        ("ProjRefEPSG", ctp.c_int),
+        # Lower-level information for GMT use only
+        ("hidden", ctp.c_void_p),
+    ]
+
+    def _parse_dimensions(self):
+        """
+        Get dimension names and attributes from the grid header.
+
+        For a 2-D grid, the dimension names are set to "y" and "x" by default. The
+        attributes for each dimension are parsed from the grid header following GMT
+        source codes. See the GMT functions "gmtnc_put_units", "gmtnc_get_units" and
+        "gmtnc_grd_info" for reference.
+        """
+        # Default dimension names.
+        dims = ("y", "x")
+        nameunits = (self.y_units, self.x_units)
+
+        # Dictionary for dimension attributes with the dimension name as the key.
+        attrs = {dim: {} for dim in dims}
+        # Dictionary for mapping the default dimension names to the actual names.
+        newdims = {dim: dim for dim in dims}
+        # Loop over dimensions and get the dimension name and attributes from header
+        for dim, nameunit in zip(dims, nameunits, strict=False):
```

Comment:
Should zip with `strict=True`
```suggestion
        for dim, nameunit in zip(dims, nameunits, strict=True):
```